File: stack/150.py
class Solution:
    def evalRPN(self, tokens):
        """
        :type tokens: List[str]
        :rtype: int
        """
        stack = []
        
        for c in tokens:
            
            if c == '+':
                temp = stack.pop()
                temp1 = stack.pop()
                stack.append(temp+temp1)
            elif c =='-':
                temp = stack.pop()
                temp1 = stack.pop()
                stack.append(temp1 - temp)
            elif c == '*':
                temp = stack.pop()
                temp1 = stack.pop()
                stack.append(temp*temp1)                
            elif c == '/':
                temp = stack.pop()
                temp1 = stack.pop()
                stack.append(int(temp1 / temp))     
            else:
                stack.append(int(c))
                
                
        return stack.pop()
            

File: stack/150.cc
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        std::stack<int> s;
        int first = 0, second = 0;
        for (const auto &token : tokens) {
            if (token == "+") {
                first = s.top();
                s.pop();
                second = s.top();
                s.pop();
                s.emplace(first + second);
            } else if (token == "-") {
                first = s.top();
                s.pop();
                second = s.top();
                s.pop();
                s.emplace(second - first);                
            } else if (token == "*") {
                first = s.top();
                s.pop();
                second = s.top();
                s.pop();
                s.emplace(first * second);                
            } else if (token == "/") {
                first = s.top();
                s.pop();
                second = s.top();
                s.pop();
                s.emplace(second / first);                 
            } else {
                s.emplace(std::stoi(token));
            }
        }
        return s.top();
    }
};
